lib/sync-metrics.js

```diff
@@ -148,28 +148,33 @@ var getStats = function(cb) {
       return input.toFixed(2) + 'ms';
     }
   }];
-  async.map(metricsToFetch, function(metric, callback) {
-    var metricName = metric.metricName;
-    redisClient.lrange([statsNamespace, metricName].join(':'), 0, MAX_NUMBER, function(err, data) {
+  if (redisClient) {
+    async.map(metricsToFetch, function(metric, callback) {
+      var metricName = metric.metricName;
+      redisClient.lrange([statsNamespace, metricName].join(':'), 0, MAX_NUMBER, function(err, data) {
+        if (err) {
+          debugError('Failed to get values from redis for key %s  with error : %s', metricName, err);
+          return callback();
+        }
+        var stats = aggregateData(metric, data);
+        stats.name = metric.displayName;
+        return callback(null, stats)
+      });
+    }, function(err, results) {
       if (err) {
-        debugError('Failed to get values from redis for key %s  with error : %s', metricName, err);
```

Comment:
We going to miss that debug in new version. (not sure if that was intentional and we debug callback bellow.
Gong to check that.